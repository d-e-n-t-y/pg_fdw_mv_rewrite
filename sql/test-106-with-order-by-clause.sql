-- FIXME: wrt ORDER BY: hmmm... this seems to be fine... but not really sure why it is working.
--        Ah. It seems to be that FDW doesn't push down ORDER BY for a GROUPed query.
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test_lb group by key order by key;

select key, COUNT (value) from test_lb group by key order by key;

explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test_lb group by key order by key desc;

select key, COUNT (value) from test_lb group by key order by key desc;
