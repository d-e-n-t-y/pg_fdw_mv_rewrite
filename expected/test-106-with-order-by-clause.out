-- FIXME: wrt ORDER BY: hmmm... this seems to be fine... but not really sure why it is working.
--        Ah. It seems to be that FDW doesn't push down ORDER BY for a GROUPed query.
explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test_lb group by key order by key;
                         QUERY PLAN                         
------------------------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test_lb.key
   ->  Foreign Scan
         Output: key, (count(value))
         Relations: Aggregate on (public.test_lb)
         Remote SQL: SELECT key, count FROM public.test_mv1
(7 rows)

select key, COUNT (value) from test_lb group by key order by key;
 key  |  count  
------+---------
 key1 |      10
 key2 | 1000000
(2 rows)

explain (VERBOSE, COSTS OFF) select key, COUNT (value) from test_lb group by key order by key desc;
                         QUERY PLAN                         
------------------------------------------------------------
 Sort
   Output: key, (count(value))
   Sort Key: test_lb.key DESC
   ->  Foreign Scan
         Output: key, (count(value))
         Relations: Aggregate on (public.test_lb)
         Remote SQL: SELECT key, count FROM public.test_mv1
(7 rows)

select key, COUNT (value) from test_lb group by key order by key desc;
 key  |  count  
------+---------
 key2 | 1000000
 key1 |      10
(2 rows)

